test/studies/shootout/reverse-complement/bharshbarg/revcomp-buf.chpl

```diff
@@ -0,0 +1,155 @@
+/* The Computer Language Benchmarks Game
+   http://benchmarksgame.alioth.debian.org/
+
+   contributed by Ben Harshbarger
+   derived from the Rust #2 version by Matt Brubeck
+*/
+
+/*
+   This is very ugly because we don't have good IO support for
+   reading/writing buffers. The 'data' buffer is included in the
+   'buf' record so that we know exactly how many bytes to read
+   with 'readBytes'. This is so that we don't accidentally read
+   past the end of the file and get an error. This would be bad
+   because 'readBytes' doesn't tell us how many bytes it actually
+   read...
+*/
+
+config param columns = 61;
+
+const table = initTable("ATCGGCTAUAMKRYWWSSYRKMVBHDDHBVNN\n\n");
+
+record buf {
+  const bufSize : int;
+  var buf : [0..#bufSize] uint(8);
+  var cur, cap, numLeft : int;
+  var chan : channel(writing=false, kind=iokind.native, locking=false);
+
+  proc init(fi:file, bs:int) {
+    this.bufSize = bs;
+
+    super.init();
+
+    chan = fi.reader(locking=false);
+    numLeft = fi.length();
+  }
+
+  pragma "no copy return"
```

Comment:
Can you put a comment saying that this returns a view into the buffer starting at low ?